stm/stm.ml

```diff
@@ -2418,18 +2472,22 @@ let finish_tasks name u d p (t,rcbackup as tasks) =
     msg_error (str"File " ++ str name ++ str ":" ++ spc () ++ iprint e);
     exit 1
 
-let merge_proof_branch ~valid ?id qast keep brname =
+let merge_proof_branch ~valid ?id qast keep brname parsing =
   let brinfo = VCS.get_branch brname in
   let qed fproof = { qast; keep; brname; brinfo; fproof } in
   match brinfo with
   | { VCS.kind = `Proof _ } ->
       VCS.checkout VCS.Branch.master;
-      let id = VCS.new_node ?id () in
+      (* Proofs should begin before Lemma, hence picking the parsing mode of
+       * the fork should be enough, but they start just after the Lemma so
+       * the proof mode is active.  Hence we explicitly pop *)
+      let parsing = Parser.unset_proof_mode parsing in
+      let id = VCS.new_node ?id parsing in
```

Comment:
It should be executed once per Qed already. I can't put the code in process transaction because the new node is built here, but it should really be the same code path.